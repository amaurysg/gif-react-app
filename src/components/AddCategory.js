import React, { useState } from 'react'
import PropTypes from 'prop-types'
import swal from 'sweetalert';
import propTypes from 'prop-types'


const AddCategory = ({setCategories}) => {


  const [inputValue, setInputValue] = useState('')

  const handleInputChange = (e)=>{  
    setInputValue(e.target.value)
    console.log("Llamando onchange")
  }

  const handleSubmit = (e)=>{
    e.preventDefault()

    if( inputValue.trim().length > 2 ){

      //This line is to test (show error without beforeEach())
      console.log('handleSubmit (for test)', inputValue)

      setCategories(cat => [inputValue,...cat] )
      setInputValue('')
     
    }else{
      swal("Oops!", "Please, add something!", "info");
    }
  }

  return (
      <>
        <form onSubmit={handleSubmit} >
          <p>{inputValue} </p>
            <input 
            type="text"
            value={inputValue}
            placeholder='Search your gifs...'
            onChange={handleInputChange}
                  
            />       
        </form>

      </>
    
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}



export default AddCategory
