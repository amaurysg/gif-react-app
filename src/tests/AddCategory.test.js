import React from 'react'
import { shallow } from "enzyme"
import AddCategory from "../components/AddCategory"

describe('Test component AddCategory.js', () => {
  
  test('Debe renderizar el component OK', () => {

      const setCategories = ()=>{}

      const wrapper = shallow(<AddCategory setCategories={setCategories} />)

      expect(wrapper).toMatchSnapshot()

  })
  

})
