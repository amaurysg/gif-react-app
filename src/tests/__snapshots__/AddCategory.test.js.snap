// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component AddCategory.js Debe renderizar el component OK 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Search your gifs..."
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
